### DISCUSSION TITLE
CRD Generator v2 minimize quotes

### DISCUSSION THREAD

**Discussion by MichaelMorrisEst** [fabric8io/kubernetes-client#6748] [General]
Im looking at moving to use v2 of the CRD generator via the maven plugin however I notice that, unlike in v1, YAMLGenerator.Feature.MINIMIZE_QUOTES is not enabled, resulting in superfluous quotes appearing in the generated yaml.
Can this be changed or at least made configurable?

**Comment by shawkins** [comment]
The same default style yaml is now being across the kubernetes client project - see https://github.com/fabric8io/kubernetes-client/blob/5fd904ced7a1798d357d41376de1e699837c185e/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesSerialization.java#L190

The only workaround currently is that programatic use of the v2 CRD logic allows for passing in a customized ObjectMapper / KubernetesSerialization, so you could override that method however you wanted.

As for making it configurable in general an enhancement could be created for that - if it's something you contribute that would be great. In theory it shouldn't need to be specific to just CRD generation, but there is some internal usage of the yamlification that needs to remain consistent.
